### ISSUE TITLE
How to pass multiple rotation values for augmentation?

### ISSUE THREAD

**Issue by SuroshAhmadZobair** [ultralytics/ultralytics#1013]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi,

How can i pass multiple values for rotation augmentation during training?
For ex, instead of `degree=45` I would like to pass the following `degree=([0,45,15,80],prob=0.3)`

Any help is appreciated!
cheers!

**Comment by kurkurzz** [comment]
Please open this issue. Is there a way to apply custom augmentations on YOLOv8

**Comment by glenn-jocher** [comment]
@kurkurzz certainly! To use custom augmentations in YOLOv8, you can integrate them directly into your dataset's processing pipeline. For example, if you're using PyTorch, you can modify your dataset class to include any transformations you'd like during the `__getitem__` method. Here’s a quick example using `albumentations`:

```python
from albumentations import Compose, RandomBrightnessContrast, HueSaturationValue
import cv2
from torch.utils.data import Dataset

class CustomYOLODataset(Dataset):
    def __init__(self, images, transforms=None):
        self.images = images  # a list of image paths
        self.transforms = transforms

    def __len__(self):
        return len(self.images)

    def __getitem__(self, idx):
        image = cv2.imread(self.images[idx])  # or any other method to load your image
        if self.transforms:
            augmented = self.transforms(image=image)
            image = augmented['image']
        # Convert your image to PyTorch tensor and perform further processing as required
        # Return your processed image and other items (like targets) as needed
        return image

# Define your custom augmentations
custom_transforms = Compose([
    RandomBrightnessContrast(p=0.2),
    HueSaturationValue(hue_shift_limit=20, sat_shift_limit=30, val_shift_limit=20, p=0.3),
])

# Create your dataset
dataset = CustomYOLODataset(images=["path/to/your/images"], transforms=custom_transforms)

# Now use this dataset for training with YOLOv8
```

This way, you can insert any custom augmentation you need for your specific use-case. Remember to ensure that the augmentations do not disrupt the integrity of your bounding box labels. Happy training! 🚀
